src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java

```diff
@@ -110,7 +111,12 @@ public void projectOpened(@NotNull Project project) {
 
         String repoPath = FileUtils.normalizeFilePath(contentRoot.getPath());
         String repoName = contentRoot.getName();
-        CodeSyncSetup.setupCodeSyncRepoAsync(project, repoPath, repoName, false, false);
+        RepoStatus repoStatus = StateUtils.getRepoStatus(repoPath);
+        if (repoStatus == RepoStatus.DISCONNECTED) {
```

Comment:
The Setup part should still run I think, but instead of showing "Repo is in sync", we can show relevant notifications

Reply:
Do we allow users to sync repo even when account is deactivated?

Reply:
Actually, in intelliJ creating directories is separate. setup just starts the repo syncing process.